packages/block-library/src/block-keyboard-shortcuts/index.js

```diff
@@ -46,19 +46,33 @@ function BlockKeyboardShortcuts() {
 		) {
 			return;
 		}
+		const newAttributes = {
+			content: attributes.content,
+		};
 
-		const textAlign =
-			blockName === 'core/paragraph' ? 'align' : 'textAlign';
-		const destinationTextAlign =
-			destinationBlockName === 'core/paragraph' ? 'align' : 'textAlign';
+		// Read textAlign from source block (could be in old or new format)
+		const sourceTextAlign =
+			attributes.textAlign || attributes.style?.typography?.textAlign;
```

Comment:
At first glance, I thought that if the source block was a paragraph block, we should check `attributes.align` instead of `attributes.textAlign`, but it seems that the migration of the paragraph block has already been processed at this point.